### ISSUE TITLE
Does not change volume for default windows driver

### ISSUE THREAD

**Issue by barbarojacp** [irzyxa/Volume2#131]
Using ASIO Link Pro (VADPRO) WDM device as my default. Does not change volume. Also, the link button seems to work only if a broken link is shown, which is confusing.

**Comment by irzyxa** [comment]
There is a hint on button with broken link.

**Comment by barbarojacp** [comment]
Yes, that's only a minor quirk. The main issue is not being able to control the main windows device volume.

**Comment by irzyxa** [comment]
What environment do you use? What devices or programs? How can I emulate your situation?

**Comment by barbarojacp** [comment]
Using W10x64. I use a WDM device (asiolink pro) as input to GigPerformer 3.
GigPerformer 3 with asio link pro can use whatever asio device I want. I
have a multimonitoring setup with GigPerformer 3.

I use windows key plus mouse wheel as volume adjust. Since there are always
minimum of 2 devices involved, is very convenient to use the link function
so both of them change volume. This would be even more useful with a
midi/OSC output capability, but that's another subject. Also, I switch
audio interfaces a lot, so having volume2 remember other linked volume
controls would be awesome too.

This kind of routing scenarios like I have are used plenty by podcasters,
music professionals and in general any kind of creator. Other hosts like
GigPerformer are liveprofessor, cantabile, voicemeeter, vsthost, and many
many others.

Thanks for helping!

**Comment by irzyxa** [comment]
Does Windows see your ASIO devices? Can you show screenshot list of devices?

**Comment by barbarojacp** [comment]
For some reason in this session it is working. It is kind of temperamental
according to my experience. Screenshot below shows the devices that should
be linked. Since I can only link one extra device, my main windows device
is ASIOVADPRO and my secondary is Realtek.

Windows does not see ASIO devices by default. Gigperformer is the one that
says what ASIO device to use. ASIO LINK PRO does have the functionality as
well but I use GIg performer to be able to use Dirac and headphone
equalizers in a studio environment.

It is fairly simple, just sounds more complicated than it really is. So I
will explain it in a simpler way as follows:

So 2 ways of controlling audio devices. Under windows, WDM and under DAWs,
ASIO. The main point of using a router driver like ASIO Link Pro is the
ability to have one central piece of software like Gigperformer to operate
every monitoring related needs you could ever imagine. So windows sound
(ie, youtube, video player, mp3) is processed, as well as a low latency
ASIO output from your DAW. Any DAW, no need to turn off plugins when
rendering.Not to mention that each headphone output can be
proccessed individually. Just genius.

So Windows (Using WDM ASIOVADPRO) > Gigperformer (Processes monitoring, and
chooses ASIO driver, like Asio4all to use realtek) > Sound
And DAW (Like Ableton Live using ASIO mode and AsioLinkPro driver) >
Gigperformer ASIO link internal input so uses the realtek as explained
above) > Sound

[image: image.png]
